src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
I think the proper way to do it is probably this:

- use async.queue with a concurrency of e.g 4
- push all addrs as task to the queue
- start the queue
- in the worker function check if success and if so call `queue.kill` and the callback
- if after finishing the queue no success happened call back in with an error